### ISSUE TITLE
broadcast issue with scipy.stats.binom.ppf (?)

### ISSUE THREAD

**Issue by twmacro** [scipy/scipy#5631]
I'm getting a "value too deep" error from `binom.ppf` where I think it should work. At least, it is not consistent with the other functions I tried. Here is a tiny script that illustrates this issue (I'm not claiming these numbers are actually meaningful -- I just made them up on the fly):

```
import numpy as np
from scipy.stats import norm, nct, chi2, binom

c = [.4, .6]
p = .5
n = [[12], [15]]
print(norm.ppf(c, n, p))
print(nct.ppf(c, n, p))
print(chi2.ppf(c, n, p))
print(binom.ppf(c, n, p))  # gets 'value too deep error'
```

The output is:

```
[[ 11.87332645  12.12667355]
 [ 14.87332645  15.12667355]]
[[ 0.25151054  0.77244159]
 [ 0.25053324  0.76855878]]
[[ 10.68197138  13.08383797]
 [ 13.5297496   16.23322295]]
Traceback (most recent call last):
  File "binom_bug.py", line 10, in <module>
    print(binom.ppf(c, n, p))  # gets 'value too deep error'
  File "/home/macro/anaconda3/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py", line 3056, in ppf
    place(output, cond2, self.b)
  File "/home/macro/anaconda3/lib/python3.5/site-packages/numpy/lib/function_base.py", line 1595, in place
    return _insert(arr, mask, vals)
ValueError: object too deep for desired array
```

**Comment by josef-pkt** [comment]
There were problems in general to vectorize for parameters that change the support of the distribution. In this case, `n` changes the support. All other parameters should work for vectorized.
None of the other distributions in your list change the support.

I think it has been fixed for some distributions, but not everywhere.

**Comment by josef-pkt** [comment]
It looks like it works if `n` is 1-D

```
>>> scipy.__version__
'0.15.1'

>>> stats.binom.ppf(np.array([[0.2, 0.5, 0.8]]).T, np.ravel(n), 0.6)
array([[  6.,   7.],
       [  7.,   9.],
       [  9.,  11.]])
```

**Comment by josef-pkt** [comment]
```
>>> stats.binom.ppf(np.array([[0, 0.5, 1]]).T, np.ravel(n), 0.6).T
array([[ -1.,   7.,  12.],
       [ -1.,   9.,  15.]])

>>> stats.binom.b
array([12, 15])
```

(I didn't remember this works.)

**Comment by twmacro** [comment]
Thank you @josef-pkt! It's nice to know there is rationale behind the behavior. :) My current routine make use of `np.broadcast` since the shape of all three inputs is up to the user.

**Comment by josef-pkt** [comment]
Because most distributions don't have changing support the bounds are just scalars -inf, 0 or inf, there is no need to broadcast the scalar to the full size of the broadcasted arguments, so it is or was not part of the generic design.

I think in this case explicit broadcasting for `b` could be added in _argcheck, maybe conditional on (size(n) > 1).

another case: cdf works also with n in column

```
>>> stats.binom.cdf(np.arange(5)[:,None], np.ravel(n), 0.6)
array([[  1.67772160e-05,   1.07374182e-06],
       [  3.18767104e-04,   2.52329329e-05],
       [  2.81018368e-03,   2.78904439e-04],
       [  1.52672666e-02,   1.92776923e-03],
       [  5.73099213e-02,   9.34766078e-03]])
>>> stats.binom.cdf(np.arange(5), np.ravel(n)[:,None], 0.6).T
array([[  1.67772160e-05,   1.07374182e-06],
       [  3.18767104e-04,   2.52329329e-05],
       [  2.81018368e-03,   2.78904439e-04],
       [  1.52672666e-02,   1.92776923e-03],
       [  5.73099213e-02,   9.34766078e-03]])

>>> stats.binom.cdf(np.arange(11, 17), np.ravel(n)[:,None], 0.6)
array([[ 0.99782322,  1.        ,  1.        ,  1.        ,  1.        ,
         1.        ],
       [ 0.9094981 ,  0.972886  ,  0.99482797,  0.99952982,  1.        ,
         1.        ]])

```

**Comment by josef-pkt** [comment]
the problem is specific to the generic code for isf and ppd

```
        place(output, (q == 0)*(cond == cond), self.a-1)
        place(output, cond2, self.b)
```

the place call doesn't work this way if self.b or self.a are arrays.
AFAICS, for cdf it works because b broadcasts in the same way as n in the conditions and is not used as "out-of-bounds" value to be `place`d inside a slice of an array.

**Comment by tqa236** [comment]
This issue seems to be fixed when I tested with version 1.4.1. Can we close it now?

**Comment by WarrenWeckesser** [comment]
@tga, thanks for checking this.  Yes, it appears to be fixed, so closing.
